base/array.jl

```diff
@@ -764,25 +764,27 @@ function findprev(A, start::Integer)
     for i = start:-1:1
         A[i] != 0 && return i
     end
-    0
+    return 0
 end
 findlast(A) = findprev(A, length(A))
 
 # returns the index of the matching element, or 0 if no matching
 function findprev(A, v, start::Integer)
     for i = start:-1:1
-        A[i] == v && return i
+        isequal(A[i], v) && return i
     end
-    0
+    return 0
 end
 findlast(A, v) = findprev(A, v, length(A))
 
 # returns the index of the previous element for which the function returns true, or zero if it never does
 function findprev(testf::Function, A, start::Integer)
     for i = start:-1:1
-        testf(A[i]) && return i
+        if testf(A[i])
```

Comment:
Why make this change here and not above? (Anyway, that's clearly unrelated to this PR...)